README.md

```diff
@@ -314,6 +314,23 @@ client = Kubeclient::Client.new(
 Note that this returns a token good for one hour. If your code requires authorization for longer than that, you should plan to
 acquire a new one, by calling `.context()` or `GoogleApplicationDefaultCredentials.token` again.
 
+#### OIDC Auth Provider
+
+If the cluster you are using has OIDC authentication enabed you can use the `openid_connect` gem to refresh
+id-tokens when they have expired.
+
+This requires the [`openid_connect` gem](https://github.com/nov/openid_connect) which is not included in
+the `kubeclient` dependencies so should be added to your own applications bundle.
+
+The OIDC Auth Provider will not perform the initial setup of your `$KUBECONFIG` file. You will need to use something
+like [`dexter`](https://github.com/gini/dexter) in order to configure the auth-provider in your `$KUBECONFIG` file.
+
+If the id-token specified in your `$KUBECONFIG` file has not expired the OIDC Auth Provider will simply use that token.
+If it has expired then the provider will refresh the token and use that.
```

Comment:
The wordings "will refresh" ... "will not automatically refresh" sounds confusing on a quick read.
I propose here saying:
> If you use `Config.context(...).auth_options` and the kubeconfig file has `user: {auth-provider: {name: gcp}}`, kubeclient will automatically obtain a token (or use `id-token` if still valid)

to me "obtain" doesn't imply ability to keep refreshing it, nor sounds as action done *to* the token so less expectation it'll write it back to the config file (the disclaimer below is still good).

The other change is becase this section talks abstractly about "the OIDC Auth Provider" but I wanted to say specifically how to invoke this (`Client` doesn't do this, `Config` does)

Reply:
(one day if we implement #393, we can reword these sections in terms of "refresh" or "renewal")

Reply:
Updated README for more clarity, let me know what you think.